packages/annotations/src/add.ts

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { UserSession } from "@esri/arcgis-rest-auth";
+
+import {
+  addFeatures,
+  IAddFeaturesRequestOptions,
+  IAddFeaturesResult
+} from "@esri/arcgis-rest-feature-service";
+
+/**
+ * Add an annotation to ArcGIS Hub. Uses authentication to derive authorship, appends a timestamp and sets a default status of "pending" to new comments by default.
+ * @param requestOptions - request options that may include authentication
+ * @returns A Promise that will resolve with response from the service after attempting to add one or more new annotations.
+ */
+export function addAnnotations(
+  requestOptions: IAddFeaturesRequestOptions
+): Promise<IAddFeaturesResult> {
+  const session = requestOptions.authentication as UserSession;
```

Comment:
TIL _a better way_ than the [`IUserRequestOptions`](https://github.com/Esri/arcgis-rest-js/blob/b144d0551c80023e13912cc92c3b7590be3da30c/packages/arcgis-rest-auth/src/authenticated-request-options.ts#L16-L21) business i rallied for earlier.

Reply:
So this works if the `authentication` is an [ApplicationSession](https://esri.github.io/arcgis-rest-js/api/auth/ApplicationSession/), right? I think you're only casting it here so TS doesn't blow up on the next line, right?